pkg/server/auth/server.go

```diff
@@ -488,6 +491,42 @@ func (s *AuthServer) UserInfo(rw http.ResponseWriter, r *http.Request) {
 	toJSON(rw, ui, s.Log)
 }
 
+// Refresh is used to refresh the access token and id token. It updates the cookies on the response
+// with the new tokens. It returns the new user principal.
+func (s *AuthServer) Refresh(rw http.ResponseWriter, r *http.Request) (*UserPrincipal, error) {
```

Comment:
if oidc is not enabled we shouldn't attempt a refresh either..

Reply:
If both oidc/user-account is enabled, and we try to refresh the user-account token, it will fail and log the user out anyway though.